crates/sqlexec/src/background_jobs.rs

```diff
@@ -0,0 +1,326 @@
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc, Mutex,
+    },
+    time::Duration,
+};
+
+use async_trait::async_trait;
+use tokio::{
+    sync::{mpsc, oneshot},
+    task::JoinHandle,
+    time::Instant,
+};
+
+use crate::errors::{ExecError, Result};
+
+/// Collection of background jobs.
+pub struct JobRunner {
+    send: mpsc::UnboundedSender<ChannelMessage>,
+    listen: Arc<Mutex<Option<JoinHandle<()>>>>,
+}
+
+struct JobHandle {
+    join: JoinHandle<()>,
+    sleep: JoinHandle<()>,
+}
+
+enum ChannelMessage {
+    NewJob(BackgroundJob),
+    JobComplete(String, Result<()>),
```

Comment:
Should I add a new message, `JobStarted`, signalling, we can safely remove the job from the map. Can maintain a `running_jobs` map to mark jobs as started but incomplete. This allows us to add new jobs with the same name when an existing job has started. Not sure if it's for the better or worse.

Reply:
@scsmithr what are your opinions on this?

Reply:
Gave it some thought, feels like the current implementation is fine. If a job is running and another job comes in that could've been batched just seconds ago, it's ok to let go of the job.

For example: if `"storage_calculation"` job is running, and another `"storage_calculation"` comes in, we already have the latest data being collected, so we can ignore the new job safely.

Reply:
I'm down to keep this as is, and then as we start building out jobs, we can reevaluate if we need a new message.